pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -186,7 +229,8 @@ func (r *Reconciler) ReconcileCNI(ctx context.Context, log logr.Logger, tinkerbe
 	return r.cniReconciler.Reconcile(ctx, log, client, clusterSpec)
 }
 
-func (r *Reconciler) validateTinkerbellIPMatch(ctx context.Context, clusterSpec *c.Spec) error {
+func (r *Reconciler) validateTinkerbellIPMatch(ctx context.Context, tinkerbellScope *Scope) error {
```

Comment:
does this actually need the scope?

Reply:
actually no, but all phases take the same input